article/_posts/2019-11-21-stealml.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: review
+title: "Stealing Machine Learning Models via Prediction APIs"
+tags: machine-learning deep-learning
+author: "Antoine Théberge"
+cite:
+    authors: "Florian Tramèr, Fan Zhang, Ari Juels, Michael K. Reiter, Thomas Ristenpart"
+    title:   "Stealing Machine Learning Models via Prediction APIs"
+    venue:   "USENIX"
+pdf: "https://arxiv.org/abs/1609.02943"
+---
+
+
+# Highlights
+- Simple equation-solving model extraction attacks for simple ML model
+- Model extraction attacks against models that output only class labels
+
+# Introduction
+
+> Machine learning (ML) models may be deemed confidential due to their sensitive training data, commercial value, or use in security applications. Increasingly often, confidential ML models are being deployed with publicly accessible query interfaces. ML-as-a-service (“predictive analytics”) systems are an example: Some charge users for access on a pay-per-query basis.
+> The tension between model confidentiality and public access motivates [the authors'] investigation of model extraction attacks. In such attacks, an adversary with black-box access, but no prior knowledge of an ML model’s parameters or training data, aims to duplicate the functionality of (i.e., “steal”) the model
+
+![](/article/images/stealml/fig1.jpeg)
+
+The goal here is to, after gaining access to a black-box target model $$f$$, train a model $$\hat{f}$$ that closely approximates or even matches $$f$$. Matching here implies that, with $$f_i(x)$$ being either the prediction over x for the _$$i^{th}$$_ class or the confidence score associated with the _$$i^{th}$$_ class, $$f_i(x) \equiv \hat{f}_i(x)$$ $$\geq$$ 99.9% of times. 
+
+![](/article/images/stealml/tab2.jpeg)
+
+# Methods and results
+
+## Data
+
+![](/article/images/stealml/tab3.jpeg)
+
+## Extraction with Confidence Values
+
+First off, extraction attacks on prediction APIs that return confidence values. Here, $$ f(x) \in [0,1]^c$$ where c is the number of classes. For simple enough models, _equation-solving attacks_ can be used to directly infer the model's parameters. For example, in the case of a binary logistic regression (LR) with parameters $$w \in R^d$$ that outputs a probability $$f_1(x) = \sigma(wx)$$, and given a oracle sample $$(x, f(x))$$, we get a linear equation $$wx = \sigma^{-1}(f_1(x))$$ and thus, only d+1 linearly independent samples can be used to recover the parameters.
+
+For all the datasets shown in table 3, the authors could train a model and extract it in d+1 predictions with 100% confidence error. For these models, the attack only required 41 queries on average and 113 at most.
```

Comment:
(...) extract it in d+1 predictions (...) : (...) extract it in $$d+1$$ predictions (...)

Reply:
fixed in a5fac43